lib/cyclonedx/report_ruby_gems_cyclonedx.rb

```diff
@@ -3,5 +3,20 @@ class ReportRubyGems < Base
     def initialize(scan_report)
       super(scan_report)
     end
+
+    def package_url(dependency)
+      "pkg:#{dependency[:type]}/#{dependency[:name]}#{version_string(dependency, true)}"
+    end
+
+    # Return version string to be used in purl
+    def version_string(dependency, is_purl_version = false)
+      prefix = is_purl_version ? "@" : ""
+      if dependency[:dependency_file] == 'Gemfile.lock'
+        # Return empty string if concrete dependency version specified in Gemfile.lock
+        "#{prefix}#{dependency[:version]}"
+      else
+        ""
+      end
+    end
```

Comment:
It would be really good to use the version of the resolved gem that is actually being used over what is in the gemfile. That's part of the purpose of CycloneDX.

Reply:
Ah, just re-read the docs on how Salus works. I don't know how you can accurately determine package versions outside of a build process.

Reply:
Yep, we're currently using the Gemfile.lock to determine the exact version but in the case of just a Gemfile that's provided we don't include the version.